packages/metamask/src/index.ts

```diff
@@ -157,8 +157,9 @@ export class MetaMask extends Connector {
             method: 'wallet_switchEthereumChain',
             params: [{ chainId: desiredChainIdHex }],
           })
-          .catch((error: ProviderRpcError) => {
-            if (error.code === 4902 && typeof desiredChainIdOrChainParameters !== 'number') {
+          .catch((error: any) => {
+            const errorCode = error.data?.originalError?.code || error.code
```

Comment:
Please provide a comment description of why this code path was added and what it solve. We didn't do our best here either, b/c there's no explanation why we check for `4902`. It would be perfect if you could cover that as well! I am thinking of a simple two-line comment.

Reply:
Done. Thanks pointing out. Sorry for not following the convention.